wgpu-core/src/device/resource.rs

```diff
@@ -2489,24 +2499,36 @@ impl Device {
                 }
 
                 let internal_use = match access {
-                    wgt::StorageTextureAccess::WriteOnly => hal::TextureUses::STORAGE_READ_WRITE,
+                    wgt::StorageTextureAccess::WriteOnly => {
+                        if !view.format_features.flags.intersects(
+                            wgt::TextureFormatFeatureFlags::STORAGE_WRITE_ONLY
+                                | wgt::TextureFormatFeatureFlags::STORAGE_READ_WRITE,
```

Comment:
Textures that support write-only all support read+write, so we don't need to check for read+write here, unless I'm missing something.

Reply:
so the logic here is checking if it has either WRITE_ONLY or READ_WRITE (intersects, not contains) and if so, then we can do write on this.

Reply:
ah, indeed, my bad!